### ISSUE TITLE
Invalid coloring in some specific cases

### ISSUE THREAD

**Issue by bramborman** [mike-ward/VSColorOutput#109]
I'm getting some wrongly-colored lines in my output on lines such as this:

````
15>Foo.cs(226,17,226,89): warning CA1806: Bar calls TryParse but does not explicitly check whether the conversion succeeded....
````

It's a warning message but probably because it contains the string _succeeded_ it gets colored to the green success-color instead of warning yellow color.

I think warning should have precedence over success maybe every time or it should be checked by what appears earlier in the string?

**Comment by mike-ward** [comment]
Probably need a separate rule that checks for a specific warning pattern. Something that matches `warning ??number:` perhaps.

**Comment by bramborman** [comment]
Well it works perfectly for other lines with the same _pattern_ that do **not** contain the word _succeeded_ so I guess the problem is that the string contains both _warning_ and _succeeded_ and the success color takes precedence. IMHO it should be that warnings have precedence.

Here is probably more obvious example:

![repro](https://user-images.githubusercontent.com/20975774/76006305-91e26100-5f0c-11ea-8854-d9671b8083e9.png)

**Comment by mike-ward** [comment]
You do know you can edit the regex's for pattern matching, right? Maybe you could try altering the regex's to get the desired result (IMHO)...

**Comment by bramborman** [comment]
Yeah, I know and I can confirm that just by reordering the original patterns this issue can be fixed. I reported it because I think it should be so by default so people like me who filter the output by color when quickly looking at it are not misled. But that's just my opinion...

**Comment by mike-ward** [comment]
Agreed. Just needed to know if that fixed your issue. So I assume just bumping it up one level fixed it. I'll add this to a new release soon.

**Comment by bramborman** [comment]
Actually had to do something more as the warning regex that needed to be moved up was the maybe second or third lowest. I don't really remember and there's no _Reset to default_ button but maybe all the warning regexes should be higher than the success ones?
